skrifa/src/outline/glyf/hint/engine/data.rs

```diff
@@ -0,0 +1,63 @@
+//! Reading and writing data.
+//!
+//! Implements 2 instructions.
+//!
+//! See <https://learn.microsoft.com/en-us/typography/opentype/spec/tt_instructions#reading-and-writing-data>
+
+use super::{Engine, OpResult};
+
+impl<'a> Engine<'a> {
+    /// Measure pixels per em.
+    ///
+    /// MPPEM[] (0x4B)
+    ///
+    /// Pushes: ppem: pixels per em (uint32)
+    ///
+    /// This instruction pushes the number of pixels per em onto the stack.
+    /// Pixels per em is a function of the resolution of the rendering device
+    /// and the current point size and the current transformation matrix.
+    ///
+    /// See <https://learn.microsoft.com/en-us/typography/opentype/spec/tt_instructions#measure-pixels-per-em>
+    /// and <https://gitlab.freedesktop.org/freetype/freetype/-/blob/57617782464411201ce7bbc93b086c1b4d7d84a5/src/truetype/ttinterp.c#L2609>
+    pub(super) fn op_mppem(&mut self) -> OpResult {
+        self.value_stack.push(self.graphics_state.ppem)
+    }
+
+    /// Measure point size.
+    ///
+    /// MPS[] (0x4C)
+    ///
+    /// Pushes: pointSize: the size in points of the current glyph (F26Dot6)
+    ///
+    /// Measure point size can be used to obtain a value which serves as the
+    /// basis for choosing whether to branch to an alternative path through the
+    /// instruction stream. It makes it possible to treat point sizes below or
+    /// above a certain threshold differently.
+    ///
+    /// See <https://learn.microsoft.com/en-us/typography/opentype/spec/tt_instructions#measure-point-size>
+    /// and <https://gitlab.freedesktop.org/freetype/freetype/-/blob/57617782464411201ce7bbc93b086c1b4d7d84a5/src/truetype/ttinterp.c#L2623>
+    pub(super) fn op_mps(&mut self) -> OpResult {
+        // Note: FreeType computes this at
+        // <https://gitlab.freedesktop.org/freetype/freetype/-/blob/57617782464411201ce7bbc93b086c1b4d7d84a5/src/truetype/ttdriver.c#L392>
+        // which is mul_div(ppem, 64 * 72, resolution) where resolution
+        // is always 72 for our purposes (Skia), resulting in ppem * 64.
```

Comment:
Isn't this potentially problemmatic for a non-Skia client? Would it make sense to store resolution, default 72, and support the mul_div?

Reply:
To be a bit more concrete, I’d expect a user to ask for a new constructor for `Size` that takes point size and resolution rather than ppem. At that point, we can plumb this through.